### DISCUSSION TITLE
strict=True breaks date parsing from json (works in pydantic)

### DISCUSSION THREAD

**Discussion by JacobHayes** [fastapi/sqlmodel#816] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from datetime import date

from pydantic import BaseModel, ConfigDict
from sqlmodel import SQLModel

json = '{"value": "1970-01-01"}'


class Pydantic(BaseModel):
    value: date


class PydanticStrict(Pydantic):
    model_config = ConfigDict(strict=True)


class SQL(SQLModel):
    value: date


class SQLStrict(SQL):
    model_config = ConfigDict(strict=True)  # validation fails


for model in [Pydantic, PydanticStrict, SQL, SQLStrict]:
    print(f"{model.__name__}: {model.model_validate_json(json)}\n")
```

### Description

SQLModels fails to validate json data with a date field into a model with `model_config = {"strict": True}`. Pydantic validates the same data correctly. For example, here's the output of the script above:

```
Pydantic: value=datetime.date(1970, 1, 1)

PydanticStrict: value=datetime.date(1970, 1, 1)

SQL: value=datetime.date(1970, 1, 1)

Traceback (most recent call last):
  File "/Users/jacobhayes/src/github.com/the-strawhecker-group/backend/x.py", line 26, in <module>
    print(f"{model.__name__}: {model.model_validate_json(json)}\n")
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jacobhayes/src/github.com/the-strawhecker-group/backend/.venv/lib/python3.12/site-packages/pydantic/main.py", line 538, in model_validate_json
    return cls.__pydantic_validator__.validate_json(json_data, strict=strict, context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
pydantic_core._pydantic_core.ValidationError: 1 validation error for SQLStrict
value
  Input should be a valid date [type=date_type, input_value='1970-01-01', input_type=str]
    For further information visit https://errors.pydantic.dev/2.6/v/date_type
```

~~I haven't confirmed but my hunch is that SQLModel might be "removing" the quotes around the date string before passing into Pydantic in this case, eg: it's sending `'1970-01-01'` instead of valid json like `'"1970-01-01"'`.~~ I think it's actually that the `model_validate` method defers to the `sqlmodel_validate` compat helper func, which just calls `cls.__pydantic_validator__.validate_python` instead of `cls.__pydantic_validator__.validate_json`. As noted in the pydantic docs, [json validation in strict mode is still a bit more flexible](https://docs.pydantic.dev/latest/concepts/strict_mode/#type-coercions-in-strict-mode) to accommodate the limited json types (eg: no native date, etc).

### Operating System

macOS

### SQLModel Version

0.0.16

### Python Version

Python 3.12.1

**Comment by JacobHayes** [comment]
I'm hoping to fix this in https://github.com/tiangolo/sqlmodel/pull/818

**Comment by ta-soft** [comment]
I wonder if there is a relationship of this to [#824](https://github.com/tiangolo/sqlmodel/discussions/824)
